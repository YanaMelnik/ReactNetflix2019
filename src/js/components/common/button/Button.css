.button {
    display: inline-block;
    padding: 5px 40px;
    font-family: Arial, serif;
    font-size: 12px;
    border-radius: 2px;
    color: white;
    font-weight: 600;
    margin: 10px 10px 30px;
    background-color: #f34c70;
    border: none;
    text-transform: uppercase;
}