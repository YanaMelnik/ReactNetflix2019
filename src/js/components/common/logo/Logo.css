.logo {
  font-weight: 700;
  color: #f34c70;
  margin-left: 30px;
  line-height: 50px;
}
