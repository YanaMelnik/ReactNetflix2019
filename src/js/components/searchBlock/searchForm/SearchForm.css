.search-film__name-title {
    display: block;
    text-transform: uppercase;
    color: white;
    font-family: Arial, serif;
    font-weight: 300;
    margin: 15px 0 20px;
}

.search-film__name-input {
    display: block;
    width: 96%;
    border: 0 solid;
    border-bottom: 2px solid #f34c70;
    background: black;
    height: 30px;
    color: white;
    font-size: 14px;
    padding-left: 20px;
}

.search-by__title {
    color: white;
    text-transform: uppercase;
    font-family: Arial, serif;
    font-size: 12px;
    margin-right: 20px;
    font-weight: 600;
}

.search-by__input {
    display: none;
}

.search-by__label {
    display: inline-block;
    background-color: #4c4a4a;
    padding: 2px 15px;
    font-family: Arial, serif;
    font-size: 12px;
    border-radius: 4px;
    color: white;
    font-weight: 400;
    margin: 10px 10px 30px;
    text-transform: uppercase;
}

.search-by input[type="radio"]:checked + label {
    background-color: #f34c70;
}

.search-by__label:hover {
    background-color: #ffc1c9;
}

.search-by__action-block {
    display: flex;
    justify-content: space-between;
}
