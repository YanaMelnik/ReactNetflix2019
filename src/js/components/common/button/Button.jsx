import React, { Component } from 'react';
import PropTypes from 'prop-types';
import 'Button.css';

class Button extends Component {
  render() {
    return (
      <button
        className='button'
        onClick={this.props.action}>
        {this.props.name}
      </button>
    );
  }
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
};

export default Button;
