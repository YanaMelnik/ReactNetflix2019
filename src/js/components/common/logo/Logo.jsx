import React from 'react';

import './Logo.css';

const Logo = () => <div
    className='logo wrapper'>
    netflixroulette
  </div>;

export default Logo;
