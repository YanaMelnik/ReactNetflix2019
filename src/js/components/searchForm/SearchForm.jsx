import React from 'react';
import Logo from '../common/logo/Logo.jsx';
import Button from '../common/button/Button.jsx';

const SearchForm = () => <div>
    <Logo/>
    <form>
      <label
        htmlFor='search-film__name'
        className='search-film__name-title'>
        Find your movie
      </label>

      <input
        id='search-film__name'
        className='search-film__name-input'/>

      <span className='search__by'>
            <span>Seach by</span>
            <span>
              <input name='search__by' type='radio' value='title' checked/>
              Title
            </span>
            <span>
              <input name='search__by' type='radio' value='gender'/>
              Gender
            </span>
          </span>
      <Button name='Search' action='' sizeBig={true}/>
    </form>
  </div>;

export default SearchForm;
