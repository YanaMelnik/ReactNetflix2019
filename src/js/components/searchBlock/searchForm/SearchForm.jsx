import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Logo from '../../common/logo/Logo';
import Button from '../../common/button/Button';
import './SearchForm.css';
import { loadAllFilms, searchFilmBy } from '../../../actions/actions';

class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.textRef = React.createRef();
    this.searchByTitleRef = React.createRef();
    this.searchByGenreRef = React.createRef();
    this.state = {
      searchBy: props.searchBy,
    };
  }

  doSearch(e) {
    this.props.history.push(`/search?text=${this.textRef.current.value}
      &searchBy=${this.searchByTitleRef.current.checked ? 'title' : 'genre'}`);
    this.props.requestFilmsList();
    const text = this.textRef.current.value;
    const searchByTitle = !!this.searchByTitleRef.current.checked;
    this.props.searchFilmCallback(text, searchByTitle ? 'title' : 'genre');
  }

  render() {
    const titleChecked = this.state.searchBy === 'title';
    const genreChecked = this.state.searchBy === 'genre';
    return <div>
      <Logo/>
      <form className='wrapper'>
        <label
          htmlFor='search-film__name'
          className='search-film__name-title'>
          Find your movie
        </label>

        <input
          id='search-film__name'
          className='search-film__name-input'
          ref={this.textRef}/>

        <div className='search-by__action-block'>
        <span className='search-by'>
        <span className='search-by__title'>Search by</span>
        <input className='search-by__input'
               id='search-by__title'
               name='search-by'
               type='radio'
               value='title'
               checked={titleChecked}
               ref={this.searchByTitleRef}
               onClick={() => this.setState({ searchBy: 'title' })}
        />
        <label className='search-by__label' htmlFor='search-by__title'>
          Title
        </label>

        <input className='search-by__input'
               id='search-by__gender'
               name='search-by'
               type='radio'
               value='gender'
               ref={this.searchByGenreRef}
               checked={genreChecked}
               onClick={() => this.setState({ searchBy: 'genre' })}
        />
        <label className='search-by__label' htmlFor='search-by__gender'>
          Gender
        </label>
        </span>
            <Button text='Search' action={e => this.doSearch(e)}/>
        </div>
      </form>
    </div>;
  }
}

SearchForm.propTypes = {
  requestFilmsList: PropTypes.func.isRequired,
  searchFilmCallback: PropTypes.func.isRequired,
  history: PropTypes.object,
  searchBy: PropTypes.string.isRequired,
};

const mapDispatchToProps = dispatch => ({
  requestFilmsList: () => dispatch(loadAllFilms()),
  searchFilmCallback: (text, type) => (dispatch(searchFilmBy(text, type))),
});


export default connect(null, mapDispatchToProps)(withRouter(SearchForm));
