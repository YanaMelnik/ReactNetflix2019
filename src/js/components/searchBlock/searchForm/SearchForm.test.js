import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';
import Logo from '../../common/logo/Logo';
import Button from '../../common/button/Button';

describe('SearchForm', () => {
    let component;
    beforeEach(() => {
        component = shallow(<SearchForm/>);
    });
    it('should render component correctly in case when all props defined', () => {
        expect(component).toMatchSnapshot();
    });

    it('should contain logo and search btn', () => {
        expect(component.find(Logo)).toHaveLength(1);
        expect(component.find(Button)).toHaveLength(1);
    });
});
