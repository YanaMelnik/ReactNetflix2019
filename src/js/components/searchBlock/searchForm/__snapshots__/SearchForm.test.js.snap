// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render component correctly in case when all props defined 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
